flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/NestedRow.java

```diff
@@ -247,7 +251,7 @@ public Decimal getDecimal(int pos, int precision, int scale) {
 	public byte[] getBinary(int pos) {
 		assertIndexIsValid(pos);
 		int fieldOffset = getFieldOffset(pos);
-		final long offsetAndLen = segments[0].getLong(fieldOffset);
+		final long offsetAndLen = SegmentsUtil.getLong(segments, fieldOffset);
```

Comment:
Do we have any tests to cover this change? i.e. is there any tests can reproduce failures if we don't fix these codes?

Reply:
Yes. They are checked in ComplexTest::testNestedRow. I'll split the test file into multiple test files.